tests/platform_tests/conftest.py

```diff
@@ -91,69 +92,91 @@ def advanceboot_loganalyzer(duthosts, rand_one_dut_hostname, request):
     yield
 
     result = loganalyzer.analyze(marker, fail=False)
-    messages = result["expect_messages"].values()
-    if not messages:
-        logging.error("Expected messages not found in syslog")
-        return
-    messages = messages[0]
-
-    service_restart_times = dict()
-    service_patterns = {
-        "Stopping": re.compile(r'.*Stopping.*service.*'),
-        "Stopped": re.compile(r'.*Stopped.*service.*'),
-        "Starting": re.compile(r'.*Starting.*service.*'),
-        "Started": re.compile(r'.*Started.*service.*')
-    }
-
-    def service_time_check(message, status):
-        time = message.split(duthost.hostname)[0].strip()
-        service_name = message.split(status + " ")[1].split()[0]
-        service_dict = service_restart_times.get(service_name, {"timestamp": {}})
-        timestamps = service_dict.get("timestamp")
-        if status in timestamps:
-            service_dict[status+" count"] = service_dict.get(status+" count", 1) + 1
-        timestamps[status] = time
-        service_restart_times.update({service_name: service_dict})
-
-    for message in messages:
-        for status, pattern in service_patterns.items():
-            if re.search(pattern, message):
-                service_time_check(message, status)
-
-    loganalyzer.save_extracted_log(dest="/tmp/log/syslog")
-    logging.info(json.dumps(service_restart_times, indent=4))
-
-    FMT = "%b %d %H:%M:%S.%f"
-    for _, timings in service_restart_times.items():
-        timestamps = timings["timestamp"]
-        timings["stop_time"] = (datetime.strptime(timestamps["Stopped"], FMT) -\
-            datetime.strptime(timestamps["Stopping"], FMT)).total_seconds() \
-                if "Stopped" in timestamps and "Stopping" in timestamps else None
-
-        timings["start_time"] = (datetime.strptime(timestamps["Started"], FMT) -\
-            datetime.strptime(timestamps["Starting"], FMT)).total_seconds() \
-                if "Started" in timestamps and "Starting" in timestamps else None
-
-        timings["reboot_time"] = (datetime.strptime(timestamps["Started"], FMT) -\
-            datetime.strptime(timestamps["Stopped"], FMT)).total_seconds() \
-                if "Started" in timestamps and "Stopped" in timestamps else None
-
-    files = glob.glob('/tmp/*-report.json')
-    if files:
-        filepath = files[0]
-        with open(filepath) as json_file:
-            report = json.load(json_file)
-            service_restart_times.update(report)
-    result = service_restart_times
-    logging.info(json.dumps(result, indent=4))
+    analyze_result = dict()
+    for key, messages in result["expect_messages"].items():
+        if "syslog" in key:
```

Comment:
May be we can move syslog and sairedis handling parts to different individual methods?

```Python
  for key, messages in result["expect_messages"].items():
      if "syslog" in key:
          analyze_syslog()
      elif "sairedis.rec" in key:
          analyze_sairedis()

```

Reply:
Moved the handling into individual methods.